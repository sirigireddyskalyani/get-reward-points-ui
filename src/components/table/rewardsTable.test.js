import React from 'react';
import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';

import RewardsTable from './rewardsTable';

describe('RewardsTable Component Snapshot', () => {
    let component;

    beforeEach(() => {
        component = renderer.create(
            <RewardsTable data={[]} />
        );
    });

    it('RewardsTable component rendered successfully', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});
describe('RewardsTable Component Behavior Testing', () => {

    it('RewardsTable should show No Records Found', () => {
        render(<RewardsTable data={[]} />);
        expect(screen.getByText(/No Records Found/)).toBeVisible();
    });

    it('RewardsTable should show records', () => {
        render(<RewardsTable data={[{
            "username": "Salma.Runte",
            "date": "2022-06-11T17:33:53.534Z",
            "expenses": 4444.99
        },
        {
            "username": "Ardith_Baumbach90",
            "date": "2022-06-17T15:12:56.849Z",
            "expenses": 3306.93
        }]} />);
        expect(screen.getByText(/Salma.Runte/)).toBeVisible();
    });
});