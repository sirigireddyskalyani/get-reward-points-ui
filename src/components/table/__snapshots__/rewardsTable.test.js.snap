// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardsTable Component Snapshot RewardsTable component rendered successfully 1`] = `
<div
  className="overflow-auto min-w-full min-h-full"
>
  <table
    className="table min-w-full min-h-full"
  >
    <thead>
      <tr>
        <th />
        <th>
          User Name
        </th>
        <th>
          Description
        </th>
        <th>
          Expense
        </th>
        <th>
          Reward Points
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan={5}
        >
          <div
            className="w-100 h-100 flex justify-center text-red"
          >
            <h3>
              No Records Found
            </h3>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
